### ISSUE TITLE
Add vertical scrolling to TileGrid

### ISSUE THREAD

**Issue by kmatch98** [adafruit/circuitpython#2848]
Per @tannewt there is currently vertical TileGrid scrolling available in the C libraries. Access to the underlying library would likely speed up vertical scrolling of a TileGrid. 

This is a request to allow python code access to the available underlying TileGrid C code for scrolling.

**Comment by tannewt** [comment]
This is what I was thinking of for the "scroll": https://github.com/adafruit/circuitpython/blob/master/shared-module/displayio/TileGrid.c#L307

Not sure what a good Python level API would be.

**Comment by kmatch98** [comment]
I’m no expert, but if you open up being able to change which tile is at the origin, then any other types of scrolling could be added as wrappers around it. 

Currently scrolling requires rearranging the tilegrid, which tends to be rather slow. If you’re mostly scrolling in one direction, then is actually faster to move a bunch of smaller tilegrids and just update the x,y locations of each. 

So anything you can give access out of the C code will likely give performance benefits. 

Maybe @deshipu would have more insight.

**Comment by tannewt** [comment]
Two other hang ups I have about exposing this are:
1) it causes the whole tile grid to refresh. Disabling autorefresh and resetting all of the indices is equivalent.
2) It'd be better to scroll at a higher level because SPI displays have a native scroll in one direction that would be much faster to use.

**Comment by kmatch98** [comment]
I misunderstood that my main performance issue was due to the autorefresh that redrew the tilegrid after each change.  I changed my code to turn autorefresh off before re-arranging all the tilegrids, followed by turning autorefresh on again.  This made a huge improvement.  

I'll close this out as a misunderstanding on my part about the performance limiting factor.  For the scrolling I was doing, the limiting factor was the auto refresh, not the redefinition of the tilegrids.
